xcdat/regridder/regrid2.py

```diff
@@ -349,49 +260,53 @@ def _map_longitude(src: xr.DataArray, dst: xr.DataArray) -> Tuple[List, List]:
     dst_west, dst_east = _extract_bounds(dst)
 
     shifted_src_west, shifted_src_east, shift = _align_axis(
-        src_west, src_east, dst_west
+        src_west,
+        src_east,
+        dst_west,
     )
 
-    mapping = []
-    weights = []
     src_length = src_west.shape[0]
+    dst_length = dst_west.shape[0]
 
-    for i in range(dst_west.shape[0]):
-        contrib = np.where(
+    mapping = [
+        np.where(
             np.logical_and(
-                shifted_src_west < dst_east[i], shifted_src_east > dst_west[i]
+                shifted_src_west < dst_east[x], shifted_src_east > dst_west[x]
             )
         )[0]
+        for x in range(dst_length)
+    ]
 
-        weight = np.minimum(dst_east[i], shifted_src_east[contrib]) - np.maximum(
-            dst_west[i], shifted_src_west[contrib]
-        )
-
-        weights.append(weight.values.reshape(1, contrib.shape[0]))
-
-        contrib += shift
+    weights = [
+        (
+            np.minimum(dst_east[x], shifted_src_east[y])
+            - np.maximum(dst_west[x], shifted_src_west[y])
+        ).reshape((1, -1))
+        for x, y in enumerate(mapping)
+    ]
```

Comment:
Same comment about adding comment to explain logic and purpose

Reply:
Fixed in https://github.com/xCDAT/xcdat/pull/533/commits/96eaae21ebdf20fe51831bca24a8a892a6ed6cf0